core/src/main/java/com/datastax/oss/driver/api/core/data/SettableById.java

```diff
@@ -59,7 +59,12 @@
   @NonNull
   @CheckReturnValue
   default SelfT setBytesUnsafe(@NonNull CqlIdentifier id, @Nullable ByteBuffer v) {
-    return setBytesUnsafe(firstIndexOf(id), v);
+    SelfT result = null;
+    for (Integer i : allIndicesOf(id)) {
+      result = (result == null ? this : result).setBytesUnsafe(i, v);
+    }
+    assert result != null; // allIndices throws if there are no results
+    return result;
```

Comment:
A couple of comments on performance:

* for bound statements (immutable), this will create a new copy for each iteration in the loop. That's fine, because repeated names are rare anyway, and if they happen it will probably be 2-3 times at most. Also `BoundStatementBuilder` is available as a workaround.
* this quirky pattern to deal with the self type is copy-pasted in every method. I could extract it to a utility method that takes a `BiFunction<SelfT, Integer, SelfT>`, but creating a capturing lambda for every method call bothers me.